packages/integration-sdk-runtime/src/execution/executeIntegration.ts

```diff
@@ -220,6 +223,45 @@ export async function executeWithContext<
       createStepGraphObjectDataUploader,
     } = options;
 
+    // Load data for ignored step dependencies.
+    if (options.useStorageForIgnoredStepDependencies) {
+      const loadedEntities: string[] = [];
+      for (const stepId of config?.dependentSteps ?? []) {
+        const path = getRootStorageAbsolutePath(`graph/${stepId}/entities`);
+        logger.info(
+          { path },
+          `Attempting to load entities for ignored step dependencies...`,
+        );
+        await walkDirectory({
+          path: path,
+          iteratee: async ({ filePath }) => {
+            const { entities } = await readGraphObjectFile<FlushedEntityData>({
+              filePath,
+            });
+            if (entities) {
+              loadedEntities.push(
+                ...entities.map((entity) => {
+                  return `Type: ${entity._type} | Class: ${entity._class} | Key: ${entity._key}`;
+                }),
+              );
+
+              await graphObjectStore.addEntities(stepId, entities);
+              stepStartStates[stepId].cacheLoaded = true;
+            }
+          },
+        });
+      }
+
+      if (loadedEntities.length > 0) {
+        logger.info(
+          { loadedEntities },
+          `Successfully loaded entities for ignored step dependencies from disk.`,
+        );
+      }
+    }
+
+    await removeStorageDirectory();
```

Comment:
No, this is because the data is exported to disk after each run. So a user flow would be: 1) `j1-integration collect -step fetch-users` and then 2) `j1-integration collection --step fetch-users -I` and 2) can be run as many times as desired.

-----

Thank you for thinking this through. I wasn't aware of how flushing worked and I think you are right.
Here are two thoughts:
1. we first copy the pertinent data to a .j1-step-data directory and then read the data in from there.
2. OR we allow the user to specify an arg via the CLI to a directory of data they want imported.

Or we implement both. If the directory arg is specified, we use it. If not, we copy the pertinent data.